### PR TITLE
[Feature] support database classifier for resource group

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Problem Summary(Required) ：

Support `database` classifier for resource group.
- The `database` has higher priority over other classifiers
- Multiple database could be specified with comma separated string
- If and only if the database list could cover the dbNames used in a query, the resource group could be matched

Usage example:
```sql
create resource_group rg1
to (database='default_cluster:hehe') 
with ( 'cpu_core_limit' = '10', 'mem_limit' = '10%')
```

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 30 / 34 (88.24%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|5|7|71.43%|
|:large_blue_circle:|com/starrocks/sql/analyzer/WorkGroupAnalyzer.java|7|9|77.78%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroup.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupClassifier.java|14|14|100.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|3|3|100.00%|

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 37 / 39 (94.87%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/WorkGroupAnalyzer.java|11|13|84.62%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroup.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupClassifier.java|14|14|100.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectContext.java|4|4|100.00%|

**Comment by silverbullet233** [comment]
If we created resource groups for each single database, and a query need to access some of them, which resource group will be used in expect?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/WorkGroupAnalyzer.java [hunk 1]
```diff
@@ -56,6 +61,17 @@ public static WorkGroupClassifier convertPredicateToClassifier(List<Predicate> p
...
+                    List<String> databases = Splitter.on(",").splitToList(value);
+                    if (CollectionUtils.isEmpty(databases)) {
+                        throw new SemanticException(
+                                String.format("Illegal classifier specifier '%s': '%s'", WorkGroup.DATABASES, eqPred));
+                    }
+                    String clusterName = ConnectContext.get() != null ? ConnectContext.get().getClusterName() : "";
+                    List<String> fullNames = databases.stream()
+                            .map(x -> ClusterNamespace.getFullName(clusterName, x))
+                            .collect(Collectors.toList());
+                    classifier.setDatabases(fullNames);
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/WorkGroupAnalyzer.java:74]: for user interface， we could use database name. but I think in our system, we should use database id. then, we needn't currentSqlDbNames in ConnectContext, and  long set performance is better than string set.

**Comment by mofeiatwork** [comment]
> If we created resource groups for each single database, and a query need to access some of them, which resource group will be used in expect?

Current database classifier uses a **contains all** strategy to match rules. A database classifier must contains all the databases the query accessed, so that it could be matched.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/WorkGroupAnalyzer.java (hunk 1 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/WorkGroupAnalyzer.java]: Good advice. I have changed it.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 41 / 44 (93.18%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/WorkGroupAnalyzer.java|15|18|83.33%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroup.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupClassifier.java|19|19|100.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|3|3|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**